tests/window_management_tests/test_minimal_window_manager.cpp

```diff
@@ -554,40 +558,152 @@ TEST_F(MinimalWindowManagerTest, can_resize_south_east_with_touch)
     EXPECT_EQ(window.size(), geom::Size(50, 50));
 }
 
-TEST_F(MinimalWindowManagerTest,  new_windows_are_inserted_above_older_windows)
+template <uint N>
+class MultipleWindowsMinimalWindowManagerTest : public MinimalWindowManagerTest
+{
+public:
+    void SetUp() override
+    {
+        MinimalWindowManagerTest::SetUp();
+        miral::WindowSpecification spec;
+        spec.size() = { geom::Width {100}, geom::Height{100} };
+        spec.depth_layer() = mir_depth_layer_application;
+
+        for (uint i = 0; i < N; i++)
+        {
+            apps[i] = open_application("app" + i);
+            windows[i] = create_window(apps[i], spec);
+        }
+    }
+
+    miral::Application apps[N];
+    miral::Window windows[N];
+};
+
+using TwoWindowsMinimalWindowManagerTest = MultipleWindowsMinimalWindowManagerTest<2>;
+using ThreeWindowsMinimalWindowManagerTest = MultipleWindowsMinimalWindowManagerTest<3>;
+
+TEST_F(ThreeWindowsMinimalWindowManagerTest, new_windows_are_inserted_above_older_windows)
+{
+    EXPECT_TRUE(is_above(windows[2], windows[1]));
+    EXPECT_TRUE(is_above(windows[1], windows[0]));
+}
+
+TEST_F(ThreeWindowsMinimalWindowManagerTest, when_a_window_is_focused_then_it_appears_above_other_windows)
+{
+    tools().select_active_window(windows[0]);
+    EXPECT_TRUE(focused(windows[0]));
+    EXPECT_TRUE(is_above(windows[0], windows[2]));
+    EXPECT_TRUE(is_above(windows[2], windows[1]));
+}
+
+TEST_F(TwoWindowsMinimalWindowManagerTest, if_active_window_is_removed_then_parent_has_focus_priority)
+{
+    miral::WindowSpecification spec;
+    spec.size() = { geom::Width {100}, geom::Height{100} };
+    spec.parent() = windows[0];
+    auto const child_window = create_window(apps[0], spec);
+
+    tools().ask_client_to_close(child_window);
+    EXPECT_TRUE(focused(windows[0]));
+    EXPECT_TRUE(is_above(windows[0], windows[1]));
+}
+
+TEST_F(ThreeWindowsMinimalWindowManagerTest, if_active_window_is_removed_then_last_focused_window_on_same_workspace_is_focused)
```

Comment:
There are no workspaces in this test body, so why are the mentioned in the test name?
```suggestion
TEST_F(ThreeWindowsMinimalWindowManagerTest, if_active_window_is_removed_then_last_focused_window_is_focused)
```